### DISCUSSION TITLE
How to enable Brotli compression for server responses?

### DISCUSSION THREAD

**Discussion by spikecodes** [hyperium/hyper#2497] [Q&A] [answered]
Hi,

I'm interested in enabling [text compression](https://web.dev/uses-text-compression) for my Hyper-powered server to improve load time.

Does Hyper natively support Brotli compression? Or would I have to use an external text compression library such as [rust-brotli](https://github.com/dropbox/rust-brotli)?

Thank you in advance!

**Comment by seanmonstar** [comment, marked as answer]
hyper does not include a Brotli implementation in it. You could probably use something like [`async-compression`](https://crates.io/crates/async-compression) to create a `Stream` that compresses your data, and make that be the body element of your `http::Response`s.

**Reply by spikecodes** [reply]
Thanks!
